server/controllers/personal/personalController.ts

```diff
@@ -1832,14 +1865,46 @@ export default class PersonalController {
       const { prisonerNumber, emailAddressId } = req.params
       const { clientToken } = req.middleware
 
-      await this.personalPageService.updateGlobalEmail(clientToken, prisonerNumber, emailAddressId, value)
+      const { emails } = await this.personalPageService.getGlobalPhonesAndEmails(clientToken, prisonerNumber)
+      const previousEmail = emails.find(email => email.id.toString() === emailAddressId).email
+      const emailEntered = value.replaceAll(' ', '').toLowerCase()
+      const valueHasChanged = emailEntered !== previousEmail
+      const isDuplicate = emails.some(email => email.email === emailEntered)
+
+      if (isDuplicate && valueHasChanged) {
+        req.flash('errors', [
+          {
+            text: 'This email address already exists for this person. Add a new email or edit the saved one',
+            href: '#email',
+          },
+        ])
+        req.flash('requestBody', JSON.stringify(req.body))
+        throw Error('Duplicate Email')
+      } else {
+        await this.personalPageService.updateGlobalEmail(clientToken, prisonerNumber, emailAddressId, emailEntered)
+      }
     }
 
-    const globalEmailCreator: TextFieldSetter = async (req, _res, _fieldData, value) => {
+    const globalEmailCreator: TextFieldSetter = async (req, res, fieldData, value) => {
       const { prisonerNumber } = req.params
       const { clientToken } = req.middleware
 
-      await this.personalPageService.createGlobalEmail(clientToken, prisonerNumber, value)
+      const { emails } = await this.personalPageService.getGlobalPhonesAndEmails(clientToken, prisonerNumber)
+      const emailEntered = value.replaceAll(' ', '').toLowerCase()
+      const isDuplicate = emails.some(email => email.email === emailEntered)
+
+      if (isDuplicate) {
+        req.flash('errors', [
+          {
+            text: 'This email address already exists for this person. Add a new email or edit the saved one',
+            href: '#email',
+          },
+        ])
+        req.flash('requestBody', JSON.stringify(req.body))
+        throw Error('Duplicate Email')
```

Comment:
Do we really want these being thrown? Isn't this going to end up getting caught here: https://github.com/ministryofjustice/hmpps-prisoner-profile/pull/1146/files#diff-45913de157ee3af6f75defc90ec353e1a358eacb8a672aa4dd1f69cc4a62c5b9R1983

And `req.flash('errors', [{ text: 'There was an error please try again' }])` being added?